test/test.py

```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/env python
+
+import os
+import sys
+import unittest
+from client import TestClient
+from server import TestServer
+
+client = None
+server = None
+
+def environ(name, default = None):
+	if not name in os.environ:
+		if default:
+			os.environ[name] = default
+		else:
+			raise EnvironmentError('environment variable {} is not defined'.format(name))
+	return os.environ[name]
+
+def setUpModule():
+	global client
+	global server
+	environ('SERVICE_ENDPOINT')
+	client = client or TestClient(host = '127.0.0.1', port = 7777, timeout = 5)
+	server = server or TestServer(host = '127.0.0.1', port = 7777, timeout = 3)
+	server.start()
+	with client.scope() as scope:
+		response = scope.send_request()
+
+def tearDownModule():
+	server.stop()
+
+class TestBase(unittest.TestCase):
+	import re
+	traceparent_name_re = re.compile(r'^traceparent$', re.IGNORECASE)
+	traceparent_format = r'^([0-9a-f]{2})-([0-9a-f]{32})-([0-9a-f]{16})-([0-9a-f]{2})$'
+	traceparent_format_re = re.compile(traceparent_format)
+
+	def get_traceparent(self, headers):
+		retval = []
+		for key, value in headers:
+			if self.traceparent_name_re.match(key):
+				retval.append((key, value))
+		self.assertEqual(len(retval), 1, 'expect one traceparent header, got more {!r}'.format(retval))
```

Comment:
this assertion also hits if there a zero traceparent headers, so the "got more" text is misleading.

Reply:
@discostu105 nice catch! Fixed.